mergify_engine/tests/unit/rules/test_rules.py

```diff
@@ -537,6 +537,57 @@ async def item(
     assert comment == {"message": "I really love Mergify", "bot_account": "AutoBot"}
 
 
+async def test_get_mergify_config_file_content_from_cache(
+    fake_repository: context.Repository,
+) -> None:
+    config = """
+    defaults:
+      actions:
+        comment:
+          bot_account: foo-bot
+        rebase:
+          bot_account: test-bot-account
+    pull_request_rules:
+      - name: ahah
+        conditions:
+        - base=main
+        actions:
+          comment:
+            message: I love Mergify
+          rebase: {}
+    """
+    client = mock.AsyncMock()
+    client.item.side_effect = [
+        github_types.GitHubContentFile(
+            {
+                "content": encodebytes(config.encode()).decode(),
+                "type": "file",
+                "path": ".github/mergify.yml",
+                "sha": github_types.SHAType("zeazeaze"),
+            }
+        ),
+    ]
+
+    fake_repository.installation.client = client
+    await fake_repository.get_mergify_config_file()
+    config_file_content_key = f"config_file_content/{fake_repository.installation.owner_id}/{fake_repository.repo['id']}"
+    cached_config_file_content = await fake_repository.installation.redis.get(
+        config_file_content_key
+    )
+    assert cached_config_file_content == encodebytes(config.encode()).decode()
+
+    await fake_repository.clear_config_file_content_cache(
+        fake_repository.installation.redis,
+        fake_repository.installation.owner_id,
+        fake_repository.repo["id"],
+    )
+
+    cached_config_file_content = await fake_repository.installation.redis.get(
+        config_file_content_key
+    )
+    assert cached_config_file_content is None
+
+
```

Comment:
I'd add a test case where you actually change the configuration file and it shows you don't use the old config file from the cache. 
It'd fail right now AFAICT.

Reply:
Indeed !

Reply:
I still think it would be great to have a test case where the config file changes or is cleared when a push event is received ! to be totally complete !